File: src/main/java/io/protobase/f7/formulas/math/COSH.java
package io.protobase.f7.formulas.math;

import io.protobase.f7.formulas.AbstractFormula;
import io.protobase.f7.formulas.FormulaName;
import io.protobase.f7.models.GridColumnRowKey;
import io.protobase.f7.utils.Converters;

import java.util.function.BiFunction;
import java.util.function.Function;

public class COSH extends AbstractFormula {
  public static FormulaName NAME = FormulaName.COS;
  public static COSH SELF = new COSH();

  public COSH() {
    super();
  }

  public COSH(Function<Object, Object> lookup, BiFunction<GridColumnRowKey, Object, Object> collateralLookup) {
    super(lookup, collateralLookup);
  }

  @Override
  public Object internal(GridColumnRowKey origin, Object... values) {
    checkLength(values.length, 1, NAME);
    return Math.cosh(Converters.toDouble(Converters.first(collateralLookup.apply(origin, values[0]))));
  }
}

File: src/main/js/formulas/math/COSH.ts
import { SheetColumnRowKey } from "../../models/common/SheetColumnRowKey";
import { Complex } from "../../models/common/Types";
import { Converters } from "../../utils/Converters";
import { AbstractFormula } from "../AbstractFormula";
import { FormulaName } from "../FormulaName";

export class COSH extends AbstractFormula {
  static SELF: COSH = new COSH();
  NAME = FormulaName.COSH;

  internal(origin: SheetColumnRowKey, ...values: Array<Complex>) {
    AbstractFormula.checkLength(values.length, 1, this.NAME);
    return Math["cosh"](
      Converters.toNumber(Converters.first(this.collateralLookup(origin, values[0])))
    );
  }
}
